lib/apis/v2/felixconfig.go

```diff
@@ -187,8 +187,8 @@ type FelixConfigurationSpec struct {
 }
 
 type ProtoPort struct {
-	Protocol string
-	Port     uint16
+	Protocol string `json:"protocol"`
```

Comment:
there they are, my favorite `json` tags.

Reply:
Don't we want `omitempty` on these?

Reply:
I think we do on the port, yes - but possibly not on the Protocol?  It depends if we want a "missing" protocol to mean TCP or whether we should just be explicit about it.

@bcreane what was the decision there?

Reply:
hmmm, actually shouldn't it be a *int since we want to be able to not assign it, or a value of 0 means unassigned (since I don't think 0 is a value port number is it?)

Reply:
actually reading the Felix code the port must be specified and I think we also want the protocol specified too.  So neither should be omitempty.